File: LeetCode/python3/566.py
class Solution:
    def matrixReshape(self, nums: List[List[int]], r: int, c: int) -> List[List[int]]:
        m, n = len(nums), len(nums[0]) if len(nums) > 0 else 0
        if m == 0 or m * n != r * c:
            return nums
        x, y = 0, 0
        ret = []
        temp = []
        for i in range(m):
            for j in range(n):
                temp.append(nums[i][j])
                y += 1
                if y == c:
                    ret.append(temp)
                    x += 1
                    y = 0
                    temp = []
        return ret

File: LeetCode/cpp/566.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>> &nums, int r, int c) {
        int n = nums.size();
        if (n == 0)
            return nums;
        int m = nums[0].size();
        if (n * m != r * c)
            return nums;
        vector<vector<int>> matrix;
        vector<int> temp;
        int x = 0, y = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                temp.push_back(nums[i][j]);
                ++y;
                if (y == c) {
                    ++x;
                    y = 0;
                    matrix.push_back(temp);
                    temp.clear();
                }
            }
        }
        return matrix;
    }
};
